src/cascadia/TerminalApp/TerminalSettings.h

```diff
@@ -15,6 +15,7 @@ Author(s):
 #pragma once
 
 #include "TerminalSettings.g.h"
+#include "../TerminalSettingsModel/IInheritable.h"

```

Comment:
(This is a much larger discussion that doesn't have to be implemented as a part of this PR. So really I'm thinking out loud.)
I'm wondering if we should move `TerminalSettings` over to TerminalSettingsModel. Hear me out:
- if we're making `TerminalSettings` `IInheritable`, might as well move `TerminalSettings` over to TSM. It's a bit weird for us to grab this random file from a separate project, right? haha
- With regards to creating a TermControl preview in Settings UI...
  - I hit a problem today. TSE cannot depend on TermApp because that'd create a circular dependency (TermApp <--> TSE). So I tried making it so that TSE builds a `TerminalSettings` by firing an event to TermApp, but then TermApp would need to somehow get the synthesized `TerminalSettings` object down to the profile page _and_ update on each change. Having `TerminalSettings` be its own project or a part of TSM makes it it so much easier to construct `TerminalSettings` and just have one readily available for a TermControl preview.

What do you think?

CC @DHowett for thoughts on this matter too. I might just be approaching the two things above completely wrong haha

Reply:
You might have a point here (but it should probably be a follow up)

Reply:
Bumping this thread for @DHowett because this is important for the terminal preview in SUI.

Reply:
(Chatted in meeting.)